packages/block-library/src/button/block.json

```diff
@@ -89,7 +89,30 @@
 				"radius": true
 			}
 		},
-		"__experimentalSelector": ".wp-block-button__link"
+		"__experimentalSelector": ".wp-block-button__link",
+		"blockStyles": {
```

Comment:
I'm not sure about this name. Block styles already means something else. Could this just be `styles` to match the name in theme.json?

Reply:
The only issue with `styles` is that it already exists as the top-level property [`styles](https://github.com/WordPress/gutenberg/blob/trunk/docs/reference-guides/block-api/block-metadata.md#block-styles) that keeps the list of block style variations. However, this might be a good opportunity to discuss whether the proposed `blockStyles` could be part of block style variations.

Reply:
Would be great to do that! @scruffian, @getdave, and I explored that today and it turned out to be quite an effort. Let's make the `blockStyles` key experimental and merge this PR to unblock other work, then let's find a way to bring it to the top-level styles property.